From 5a4610628334b1e02f87b7e6a8946b59b65a1ec1 Mon Sep 17 00:00:00 2001
From: Jan Friesse <jfriesse@redhat.com>
Date: Tue, 5 Feb 2013 11:32:07 +0100
Subject: [PATCH] Put handle to hdb in dispatch on unknown message

When we receive unknown dispatch type, we should put handle to hdb,
because it was correctly obtained. This problem is only cfg and confdb
libraries, cpg, evs, quorum, ... are not affected and works correctly.

Signed-off-by: Jan Friesse <jfriesse@redhat.com>
Reviewed-by: Fabio M. Di Nitto <fdinitto@redhat.com>
---
 lib/cfg.c    |    3 +--
 lib/confdb.c |    3 +--
 2 files changed, 2 insertions(+), 4 deletions(-)

diff --git a/lib/cfg.c b/lib/cfg.c
index a3aee9c..af3b675 100644
--- a/lib/cfg.c
+++ b/lib/cfg.c
@@ -212,7 +212,7 @@ corosync_cfg_dispatch (
 		default:
 			coroipcc_dispatch_put (cfg_instance->handle);
 			error = CS_ERR_LIBRARY;
-			goto error_nounlock;
+			goto error_put;
 			break;
 		}
 		coroipcc_dispatch_put (cfg_instance->handle);
@@ -227,7 +227,6 @@ corosync_cfg_dispatch (
 
 error_put:
 	(void)hdb_handle_put (&cfg_hdb, cfg_handle);
-error_nounlock:
 	return (error);
 }
 
diff --git a/lib/confdb.c b/lib/confdb.c
index 33e593d..7b045e3 100644
--- a/lib/confdb.c
+++ b/lib/confdb.c
@@ -418,7 +418,7 @@ cs_error_t confdb_dispatch (
 			default:
 				coroipcc_dispatch_put (confdb_inst->handle);
 				error = CS_ERR_LIBRARY;
-				goto error_noput;
+				goto error_put;
 				break;
 		}
 		coroipcc_dispatch_put (confdb_inst->handle);
@@ -433,7 +433,6 @@ cs_error_t confdb_dispatch (
 
 error_put:
 	(void)hdb_handle_put (&confdb_handle_t_db, handle);
-error_noput:
 	return (error);
 }
 
-- 
1.7.1

